src/passes/OptimizeInstructions.cpp

```diff
@@ -133,6 +133,69 @@ struct LocalScanner : PostWalker<LocalScanner> {
   }
 };
 
+// perform some final optimizations
+struct FinalOptimizer
+  : public WalkerPass<
+      PostWalker<FinalOptimizer, UnifiedExpressionVisitor<FinalOptimizer>>> {
```

Comment:
this does not need to be a Pass. It just needs to do a walk, so it can be like in RemoveUnusedBrs,
```cpp
struct FinalOptimizer : public PostWalker<FinalOptimizer> {
```

Reply:
I tried this but unfortunately this not works for me (visitExpression just not called) so I switched to `WalkerPass<PostWalker<>>`

Reply:
Hmm, but now with `visitBinary` it works!